src/tarantool_schema.c

```diff
@@ -590,27 +601,26 @@ tarantool_schema_get_fid_by_string(
 		struct tarantool_schema *schema_obj, uint32_t sid,
 		const char *field_name, uint32_t field_name_len
 ) {
-	struct mh_schema_space_t *schema = schema_obj->space_hash;
-	struct schema_key space_key = {
-		(void *)&sid,
-		sizeof(uint32_t), 0
-	};
-	mh_int_t space_slot = mh_schema_space_find(schema, &space_key, NULL);
-	if (space_slot == mh_end(schema))
+	struct mh_schema_space_t *sschema = schema_obj->space_hash;
+	struct schema_key space_key = { (void *)&sid, sizeof(uint32_t), 0 };
+	mh_int_t space_slot = mh_schema_space_find(sschema, &space_key, NULL);
+	if (space_slot == mh_end(sschema))
 		return -1;
-	const struct schema_space_value *space = *mh_schema_space_node(schema,
+	const struct schema_space_value *space = *mh_schema_space_node(sschema,
 			space_slot);
-	int i = 0;
-	for (i = 0; i < space->schema_list_len; ++i) {
-		struct schema_field_value *val = &space->schema_list[i];
-		if (strncmp(val->field_name, field_name, field_name_len) == 0)
-			return val->field_number;
-	}
-	return -1;
```

Comment:
It seems it was the singleton use of the schema_list/schema_list_len, so these fields can be removed now from struct schema_space_value?